### PR TITLE
Feature/#20

### PR DESCRIPTION
**Pull request by Umarali001**

### CODE REVIEW

File: src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Infrastructure/Accounts/Services/AccountServise.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        private readonly IPasswordHasherService _passwordHasherService;
+
+
+        
+        public AccountServise(IDataContext appDataContext, IUserCredentialsService userCredentialsService, IValidationService validationService, UserCredentials userCredentials, IPasswordHasherService passwordHasherService)
+        {
+            _appDataContext = appDataContext;
+            _userCredentialsService = userCredentialsService;
+            _validationService = validationService;
+            _userCredentials = userCredentials;
```
Comment (Habiba67y) [inline src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Infrastructure/Accounts/Services/AccountServise.cs]: bu kerakmas

File: src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Infrastructure/Accounts/Services/AccountServise.cs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        private readonly UserCredentials _userCredentials;
+        private readonly IPasswordHasherService _passwordHasherService;
+
+
+        
+        public AccountServise(IDataContext appDataContext, IUserCredentialsService userCredentialsService, IValidationService validationService, UserCredentials userCredentials, IPasswordHasherService passwordHasherService)
+        {
+            _appDataContext = appDataContext;
+            _userCredentialsService = userCredentialsService;
+            _validationService = validationService;
```
Comment (Habiba67y) [inline src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Infrastructure/Accounts/Services/AccountServise.cs]: validation service user service ishlatadi

File: src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Infrastructure/Accounts/Services/AccountServise.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            _userCredentials = userCredentials;
+            _passwordHasherService = passwordHasherService;
+            
+        }
+
+        public async ValueTask<UserCredentials> RegisterUserAsync(string firstName, string lastName, string emailAddress, string password, CancellationToken cancellationToken = default)
```
Comment (Habiba67y) [inline src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Infrastructure/Accounts/Services/AccountServise.cs]: ValuteTask ni ichida UserCredentails emas User bo'lishi kerak

File: src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Infrastructure/Accounts/Services/AccountServise.cs [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+          
+            return userCredentials;
+        }
+
+
+
+        public async ValueTask<User> LoginAsync(string emailAddress, string password)
+        {
+            var user = _appDataContext.Users.FirstOrDefault(user => user.EmailAddress == emailAddress);
```
Comment (Habiba67y) [inline src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Infrastructure/Accounts/Services/AccountServise.cs:45]: equals ishlatsez yaxshiroq

File: src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Infrastructure/Accounts/Services/AccountServise.cs [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+            {
+                throw new IncorrectPasswordException("Incorrect password.");
+            }
+
+            return user;
+        }
+
+
+
+        public async ValueTask<User> LogoutDeleteAsync(Guid userId)
```
Comment (Habiba67y) [inline src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Infrastructure/Accounts/Services/AccountServise.cs]: log out shart emas bu methodni o'chirib qo'ysezam bo'ladi

File: src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Infrastructure/Accounts/Services/AccountServise.cs [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+
+
+namespace Training.TruckWorld.Backend.Infrastructure.Accounts.Services
+{
+    internal class AccountServise : IAccountService
+    {
+        private readonly IDataContext _appDataContext;
+        private readonly IUserCredentialsService _userCredentialsService;
+        private readonly UserCredentials _userCredentials;
```
Comment (Habiba67y) [inline src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Infrastructure/Accounts/Services/AccountServise.cs]: nimaga yana UserCredentials modelini beryapsiz service izga tushunmadim

File: src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Infrastructure/Accounts/Services/AccountServise.cs [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+          
+            return user;
+        }
+
+
+
+        public async ValueTask<User> LoginAsync(string emailAddress, string password)
+        {
+            var user = _appDataContext.Users.FirstOrDefault(user => user.EmailAddress == emailAddress);
```
Comment (Habiba67y) [inline src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Infrastructure/Accounts/Services/AccountServise.cs:45]: equals ishlatsez yaxshi

**Review by Mirzohid-Mominov — DISMISSED** [review]

**Pull request closed without merging**